.github/workflows/codespell.yml

```diff
@@ -0,0 +1,25 @@
+# Codespell configuration is within .codespellrc
+---
+name: Codespell
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Annotate locations with typos
+        uses: codespell-project/codespell-problem-matcher@v1
+      - name: Codespell
+        uses: codespell-project/actions-codespell@v2
```

Comment:
This one is a bit better:

https://github.com/codespell-project/codespell-problem-matcher

Reply:
Sorry, what do you mean, since it is the one listed above

Reply:
Copy/paste error: that should have been https://github.com/codespell-project/actions-codespell. It has 21 forks and 80 stars, so it feels like at least some folks are keeping an eye on it.

Reply:
I think there is still remaining confusion somewhere.  We have two actions and they are complimentary:

- `codespell-project/codespell-problem-matcher@v1` -- this one is just to annotate in the github "files" view where typos are observed. handy but not mandatory, not as popular, no recent developments
- `codespell-project/actions-codespell@v2` -- the actual action to run `codespell` itself. 

so I would recommend to keep both. First first one makes no sense without the 2nd one

Reply:
I am opting to keep both. I am saying that instead of referencing the actions by tag (`@v1`, `@v2`), I would like to reference them by full commit hash so the behavior of the action cannot change out from under us.

Reply:
ah, gotcha -- I just pushed replacing with commit hexshas